src/main/scala/viper/gobra/frontend/info/implementation/typing/MiscTyping.scala

```diff
@@ -51,12 +51,12 @@ trait MiscTyping extends BaseTyping { this: TypeInfoImpl =>
   private[typing] def actualMiscType(misc: PActualMisc): Type = misc match {
 
     case PRange(exp) => exprType(exp) match {
-      case ArrayT(_, elem) => InternalSingleMulti(elem, InternalTupleT(Vector(IntT(config.typeBounds.Int), elem)))
-      case PointerT(ArrayT(_, elem)) => InternalSingleMulti(elem, InternalTupleT(Vector(IntT(config.typeBounds.Int), elem)))
-      case SliceT(elem) => InternalSingleMulti(elem, InternalTupleT(Vector(IntT(config.typeBounds.Int), elem)))
-      case GhostSliceT(elem) => InternalSingleMulti(elem, InternalTupleT(Vector(IntT(config.typeBounds.Int), elem)))
-      case MapT(key, elem) => InternalSingleMulti(key, InternalTupleT(Vector(key, elem)))
-      case ChannelT(elem, ChannelModus.Recv | ChannelModus.Bi) => elem
+      case ArrayT(_, elem) => InternalTupleT(Vector(IntT(config.typeBounds.Int), elem))
+      case PointerT(ArrayT(_, elem)) => InternalTupleT(Vector(IntT(config.typeBounds.Int), elem))
+      case SliceT(elem) => InternalTupleT(Vector(IntT(config.typeBounds.Int), elem))
+      case GhostSliceT(elem) => InternalTupleT(Vector(IntT(config.typeBounds.Int), elem))
+      case MapT(key, elem) => InternalTupleT(Vector(key, elem))
+      case ChannelT(elem, ChannelModus.Recv | ChannelModus.Bi) => InternalTupleT(Vector(elem))
```

Comment:
why did you change these types? I think this will just make your life harder

Reply:
Initially I tried to use multiAssignableTo but in the case where I have one identifier, that tried to assign the whole tuple returned by the range expression. I just wanted to easily extract as many elements as I want to account easily for that case.